samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -483,24 +490,14 @@ object SamzaContainer extends Logging {
 
       val taskName = taskModel.getTaskName
 
-      val taskObj = if (taskFactory != null) {
-        debug("Using task factory to create task instance")
-        taskFactory match {
+      debug("Using task factory to create task instance")
+      val task =
+        finalTaskFactory match {
```

Comment:
Got it. I wonder why didn't use a marker interface or typed class for this scenario. It is making all interactions with taskfactory very messy.